### PR TITLE
Add redundancy to the traffic schedule node: Add necessary messages

### PR DESCRIPTION
**Pull request by gbiggs**
### Implemented feature

This PR adds new messages required to support redundancy in the traffic schedule node (see [open-rmf/rmf_ros2!61](https://github.com/open-rmf/rmf_ros2/pull/61) for the main part of the implementation).

Resolves open-rmf/rmf#57.

### Implementation description

The messages are:

- a heartbeat (empty; the presence of the topic is sufficient),
- a fail over event notification to notify anyone interested that the backup node has been activated upon detecting the death of the primary node (empty), and
- a message that can hold all the currently-registered queries and their IDs and subscriber counts, for use in kick-starting the backup schedule node.

### CODE REVIEW

**Review by mxgrey — CHANGES_REQUESTED** [review]
I have some concerns about the potential bandwidth usage of `MirrorUpdate`. I think a relatively modest tweak to our safety checking approach can eliminate a lot of unnecessary bandwidth.

File: rmf_traffic_msgs/msg/MirrorUpdate.msg [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # The ID of the query description this update is for
 uint64 query_id
 
+# The query itself
+ScheduleQuery query
```
[the remark this answers is not in the document]
Comment (gbiggs) [reply rmf_traffic_msgs/msg/MirrorUpdate.msg:5]: Although `...registered_queries` is latched, it's not broadcast the first time until a query is registered or unregistered. This means that (currently) the new schedule node won't broadcast it for a while, probably. I guess the solution there is for the initialisation path used when it's resuming from a fail-over to do an initial broadcast of its known queries.

Adding an ID number to instances of schedule nodes is a reasonable idea as a way to know that a query update came from a new instance, but mirrors already get informed that a fail-over event has occurred and so could just stash all updates until the new `registered_queries` comes in. I'm a little wary of adding machinery to stash updates, though. It seems like we're adding increasingly complex machinery to handle increasingly unlikely corner cases.

[the remark this answers is not in the document]
Comment (mxgrey) [reply rmf_traffic_msgs/msg/MirrorUpdate.msg:5]: > I guess the solution there is for the initialisation path used when it's resuming from a fail-over to do an initial broadcast of its known queries.

Yep, I think publishing the initial set of queries should be part of the startup procedure for the schedule node. The latest message on the topic should always reflect the latest state of the schedule node's registered queries.

**Review by mxgrey — APPROVED** [review]
We'll add a schedule node versioning system in an upcoming PR.

File: rmf_traffic_msgs/msg/MirrorUpdate.msg (hunk 1 above)
[the remark this answers is not in the document]
Comment (gbiggs) [reply rmf_traffic_msgs/msg/MirrorUpdate.msg:5]: [Issue here](https://github.com/open-rmf/rmf/issues/75).

**Pull request merged**